crates/ruff/src/rules/flake8_pyi/rules/type_comment.rs

```diff
@@ -7,14 +7,15 @@ use ruff_diagnostics::{Diagnostic, Violation};
 use ruff_macros::{derive_message_formats, violation};
 
 /// ## What it does
-/// Checks for the use of type comments (e.g., `x = 1  # type: int`) in stub
-/// files.
+/// Checks for the use of type comments (e.g., `x = 1  # type: int`).
 ///
 /// ## Why is this bad?
 /// Stub (`.pyi`) files should use type annotations directly, rather
 /// than type comments, even if they're intended to support Python 2, since
 /// stub files are not executed at runtime. The one exception is `# type: ignore`.
 ///
+/// It should not be used in `.py` files either in recent versions of Python.
```

Comment:
Can we be more precise about what *recent versions* means? E.g. 3.10+? Should we only check for type comments in `py` files if the configured python version is as new as *recent version*?

Reply:
Modern type annotations are available since Python 3.6 so all version ruff support should check it.